### PR TITLE
Addded receipe for wpeframework-rdkservices to compile plugins: deviceinfo, monitor, locationsync, tracing

### PR DESCRIPTION
**Pull request by neeraj-deshpande**

### CODE REVIEW

File: recipes-wpe/wpeframework/wpeframework-rdkservices/0001-remove-default-services.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff -ruN wpeframework-rdkservices.orig/CMakeLists.txt wpeframework-rdkservices/CMakeLists.txt
+--- wpeframework-rdkservices.orig/CMakeLists.txt	2020-08-18 18:24:13.045349426 +0530
++++ wpeframework-rdkservices/CMakeLists.txt	2020-08-18 18:23:26.896163131 +0530
+@@ -23,7 +23,7 @@
+ # are located in the cmake directory. Include it in the search.
+ list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/")
+ 
+-include(services.cmake)
++# include(services.cmake)
```
[the remark this answers is not in the document]
Comment (neeraj-deshpande) [reply recipes-wpe/wpeframework/wpeframework-rdkservices/0001-remove-default-services.patch]: Done.

File: recipes-wpe/wpeframework/wpeframework-rdkservices_git.bb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    -DPLUGIN_USER_PREFERENCES=OFF \
+    -DPLUGIN_SYSTEMSERVICES=OFF \
+    -DPLUGIN_CONTROLSERVICE=OFF \
+"
+
+# ----------------------------------------------------------------------------
+
+FILES_SOLIBSDEV = ""
+FILES_${PN} += "${libdir}/wpeframework/plugins/*.so ${libdir}/*.so ${datadir}/WPEFramework/*"
+FILES_${PN} += "${includedir}/WPEFramework/*"
```
[the remark this answers is not in the document]
Comment (neeraj-deshpande) [reply recipes-wpe/wpeframework/wpeframework-rdkservices_git.bb]: Removed.

File: recipes-wpe/wpeframework/wpeframework-rdkservices_git.bb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                                 -DPLUGIN_YOUTUBE_MEMORYLIMIT=614400 \
+                                 -DPLUGIN_NETFLIX_MEMORYLIMIT=307200 \
+                                ,-DENABLE_MONITOR=OFF,"
+PACKAGECONFIG[tracing]        = "-DENABLE_TRACE_CONTROL=ON,-DENABLE_TRACE_CONTROL=OFF,"
+
+# ----------------------------------------------------------------------------
+
+EXTRA_OECMAKE += " \
+    -DBUILD_REFERENCE=${SRCREV} \
+    -DBUILD_SHARED_LIBS=ON \
```
[the remark this answers is not in the document]
Comment (neeraj-deshpande) [reply recipes-wpe/wpeframework/wpeframework-rdkservices_git.bb:59]: Here BUILD_SHARED_LIBS may not be required as type of library is specified explicitly in all the plugins. I kept it as it is also there in wpeframework-plugins recipe.

File: recipes-wpe/wpeframework/wpeframework-rdkservices_git.bb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                                 -DPLUGIN_WEBKITBROWSER_MEMORYLIMIT=614400 \
+                                 -DPLUGIN_YOUTUBE_MEMORYLIMIT=614400 \
+                                 -DPLUGIN_NETFLIX_MEMORYLIMIT=307200 \
+                                ,-DENABLE_MONITOR=OFF,"
+PACKAGECONFIG[tracing]        = "-DENABLE_TRACE_CONTROL=ON,-DENABLE_TRACE_CONTROL=OFF,"
+
+# ----------------------------------------------------------------------------
+
+EXTRA_OECMAKE += " \
+    -DBUILD_REFERENCE=${SRCREV} \
```
[the remark this answers is not in the document]
Comment (neeraj-deshpande) [reply recipes-wpe/wpeframework/wpeframework-rdkservices_git.bb:58]: wpeframework-plugins recipe is also passing  BUILD_REFERENCE in the same way. Can you elaborate?

File: recipes-wpe/wpeframework/wpeframework-rdkservices_git.bb (hunk 3 above)
[the remark this answers is not in the document]
Comment (HaseenaSainul) [reply recipes-wpe/wpeframework/wpeframework-rdkservices_git.bb:59]: Ok, then I think it is unwanted in wpeframework-plugins as well, so we can remove as like a cleanup. Will update wpeframework-plugins later

File: recipes-wpe/wpeframework/wpeframework-rdkservices_git.bb (hunk 4 above)
[the remark this answers is not in the document]
Comment (HaseenaSainul) [reply recipes-wpe/wpeframework/wpeframework-rdkservices_git.bb:58]: Ok, just keep it for the time being, but seems its value also not considered any where in the thunder plugins

**Review by HaseenaSainul — COMMENTED** [review]
Just remove  -DBUILD_SHARED_LIBS=ON  since it is a redundant thing

**Review by HaseenaSainul — APPROVED** [review]
LGTM

**Pull request merged**